File: csharp/WorkOOP/QGon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WorkOOP
{
    // Реализация класса QGon – четырёхугольник.
    internal class QGon : NGon
    {
        /*
         * QGon – конструктор, описывающий инициализацию объекта. 
         * 
         * На вход принимает массив точек в порядке обхода по четырёхугольнику.
         */
        public QGon(Point2D[] p) : base(p) { }

        /*
         * Переопределение под четырехугольник функции square()
         */
        new public double square() {
            Point2D[] a = new Point2D[] { p[0], p[1], p[2] };
            Point2D[] b = new Point2D[] { p[2], p[3], p[0] };
            return new TGon(a).square() + new TGon(b).square();
        } 
    }
}

File: python/src/core/qgon.py
from core.ngon import NGon
from core.point2d import Point2D
from core.tgon import TGon

class QGon(NGon):
	def __init__(self, p):
		self.n = 4
		self.p = p

	def __str__(self):
		st = f"QGon: "
		for i in range(len(self.p)):
			st += f"{str(self.p[i])}, "
		st = st[:len(st)-2]
		return st

	def square(self):
		a = [self.p[0], self.p[1], self.p[2]]
		b = [self.p[2], self.p[3], self.p[0]]
		return TGon(a).square() + TGon(b).square()